### ISSUE TITLE
You do not use NativeScaler and ModelEma for training?

### ISSUE THREAD

**Issue by wangpichao** [danczs/Visformer#1]
You do not use NativeScaler and ModelEma for training?

**Comment by danczs** [comment]
Yeah, we do not use amp,  beacause it causes NAN values sometimes. Thus we removed amp and Scaler in our code. See also in deit issue: https://github.com/facebookresearch/deit/issues/29.  Without this acceleration，our code takes more time during training than Deit. But amp can be freely used after training, so the the inference is as efficient.  
For ModelEma, we do not use it as in Deit.
Thanks for your question.

**Comment by m-aliabbas** [comment]
Hello, I am getting the error. 
NameError: name 'model_ema' is not defined
How can I train the model without it. Please help

**Comment by danczs** [comment]
> Hello, I am getting the error.
> NameError: name 'model_ema' is not defined
> How can I train the model without it. Please help

Sorry about this error. The code of Scaler and model_ema is not cleared completely .  We have fixed it by removing the code in main.py(ling 287,288)
